core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/ClusterBootstrapper.java

```diff
@@ -359,6 +360,17 @@ public void bootstrapNewNodes(Long stackId, Set<String> upscaleCandidateAddresse
         }
     }
 
+    private void cleanupOldSaltState(List<GatewayConfig> allGatewayConfigs, Set<Node> nodes, Set<Node> allNodes) throws CloudbreakOrchestratorFailedException {
+        List<GatewayConfig> gatewayTargets = allGatewayConfigs.stream()
+                .filter(gc -> nodes.stream().anyMatch(n -> gc.getPrivateAddress().equals(n.getPrivateIp())))
```

Comment:
What happens if the IP address happens to be reused after a repair? E.g. in Azure, I have seen that frequently a new VM that is being repaired will get the same IP address as the old VM had.

Reply:
So we're looking for the repairing gateway in this line.  The gateway configs already have been updated with the new ip and nodes (nodes to be repaired) will match.  So if it's changed or is the same, we'll select them correctly.